### PR TITLE
implementations in the rest api

### PR DESCRIPTION
**Pull request by gilsonjoanelo**
I created some implementations in the rest api, I am using it in a personal project and I missed some functions that have in the SOAP api and I implemented some in the new api, among them the possibility of searching a question through its summary. Another improved point was the ability to remove or add a version of a given project.

### CODE REVIEW

**Comment by gilsonjoanelo** [comment]
thinking in this way, it is not a good idea, because more than one record can come, it only returns a single record if it is exactly the same. But the steps of changing and removing project versions can exist right? I did the search for the summary due to an integration that we have in our project, but thinking about your vision, I realized that it is not the best practice to adopt, I will review the code.

**Comment by atrol** [comment]
>  But the steps of changing and removing project versions can exist right?

I think so, but would like to get some feedback from @vboctor

I created [Issue 30415](https://www.mantisbt.org/bugs/view.php?id=30415) to track this change.
The issue must be referenced in the commit message via `Fixes #30415`

File: core/commands/VersionUpdateCommand.php [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+            # viewing, override the current project. This to avoid problems with
+            # categories and handlers lists etc.
+            global $g_project_override;
+            $g_project_override = $t_project_id;
+        }
+        
+        $t_name = trim( $this->payload( 'name' ) );
+        $t_description = trim( $this->payload( 'description', '' ) );
+        $t_released = $this->payload( 'released', false );
+        $t_obsolete = $this->payload( 'obsolete', false );
```
Comment (dregad) [inline core/commands/VersionUpdateCommand.php:79]: Providing a default value will potentially result in unexpectedly changing the target field if it is not provided by the client in the payload.

In my opinion, in an Update operation the command should only update the information provided in the payload, and leave other fields unchanged. @vboctor what do you think ?

File: core/commands/VersionUpdateCommand.php [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+        
+        $t_timestamp = $this->payload( 'timestamp', '' );
+        if( is_blank( $t_timestamp ) ) {
+            $t_timestamp = null;
+        } else {
+            $t_timestamp = strtotime( $t_timestamp );
+        }
+        
+        $t_version = version_get($t_id);
```
Comment (dregad) [inline core/commands/VersionUpdateCommand.php:88]: Coding guidelines

File: core/commands/VersionRemoveCommand.php [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+			# viewing, override the current project. This to avoid problems with
+			# categories and handlers lists etc.
+			global $g_project_override;
+			$g_project_override = $t_project_id;
+		}
+
+		$t_id = trim( $this->payload( 'version_id' ) );
+		//$t_idNew = trim( $this->payload( 'idNew' ) );
+		
+		$t_version_id = version_remove($t_id, null);
```
Comment (dregad) [inline core/commands/VersionRemoveCommand.php:78]: Coding guidelines

File: core/commands/VersionUpdateCommand.php [hunk 4]
```diff
@@ -0,0 +1,99 @@
...
+        $t_version->version = $t_name;
+        $t_version->description = $t_description;
+        $t_version->released = $t_released? VERSION_RELEASED : VERSION_FUTURE;
+        $t_version->obsolete = $t_obsolete;
+        
+        $t_version_id = version_update($t_version);
```
Comment (dregad) [inline core/commands/VersionUpdateCommand.php:94]: Coding guidelines

**Review by dregad — CHANGES_REQUESTED** [review]
I second what @atrol said about searching for Issues by Summary; mc_issue_get_id_from_summary() is is flawed, because if there is more than 1 match for the given summary, it will just return the first one found. Therefore I also don't think this should be implemented in REST API as-is, and changes to *issues_rest.php* should be removed from this PR.

Some code review comments follow regarding the Version update/delete endpoints.

NB: I did not actually test the code yet.

**Comment by dregad** [comment]
@gilsonjoanelo are you planning to address the review comments ? If not, I'll close the PR.

**Comment by vboctor** [comment]
See related PR #1883

**Pull request closed without merging**